programs/drift/src/controller/orders.rs

```diff
@@ -1016,18 +1016,47 @@ fn sanitize_maker_order<'a>(
         Some(maker_stats)
     };
 
+    let market = perp_market_map.get_ref(&taker_order.market_index)?;
+
     let maker_order_id = maker_order_id.ok_or(ErrorCode::MakerOrderNotFound)?;
     let maker_order_index = match maker.get_order_index(maker_order_id) {
         Ok(order_index) => order_index,
         Err(_) => {
             msg!("Maker has no order id {}", maker_order_id);
-            return Ok((None, None, None, None));
+            let fallback_order_index = find_fallback_maker_order(
+                &maker,
+                match taker_order.direction {
+                    PositionDirection::Long => &PositionDirection::Short,
```

Comment:
why do we pass in a reference for position direction and market type here instead of not using references?